system-tests-new/psyclone/test_psyclone.py

```diff
@@ -0,0 +1,174 @@
+# ##############################################################################
+#  (c) Crown copyright Met Office. All rights reserved.
+#  For further details please refer to the file COPYRIGHT
+#  which you should have received as part of this distribution
+# ##############################################################################
+import filecmp
+import shutil
+from os import unlink
+from pathlib import Path
+from typing import Tuple
+from unittest import mock
+
+import pytest
+
+from fab.build_config import BuildConfig
+from fab.parse.x90 import X90Analyser, AnalysedX90
+from fab.steps.find_source_files import FindSourceFiles
+from fab.steps.grab.folder import GrabFolder
+from fab.steps.preprocess import fortran_preprocessor
+from fab.steps.psyclone import make_parsable_x90, Psyclone, psyclone_preprocessor, tool_available, MpPayload
+from fab.util import file_checksum
+
+SAMPLE_KERNEL = Path(__file__).parent / 'kernel.f90'
+
+# this x90 has "name=" keywords and is not parsable fortran
+SAMPLE_X90 = Path(__file__).parent / 'algorithm.x90'
+
+# this is the sanitised version, with the name keywords removed, so it is parsable fortran
+EXPECT_PARSABLE_X90 = Path(__file__).parent / 'expect.parsable_x90'
+
+# the name keywords which are removed from the x90
+NAME_KEYWORDS = ['name a', 'name b', 'name c', 'name d', 'name e', 'name f']
+
+
+def test_make_parsable_x90(tmp_path):
+    # turn x90 into parsable fortran by removing the name keyword from calls to invoke
+    grab_x90_path = SAMPLE_X90
+    input_x90_path = tmp_path / grab_x90_path.name
+    shutil.copy(grab_x90_path, input_x90_path)
+
+    parsable_x90_path = make_parsable_x90(input_x90_path)
+
+    # the point of this function is to make the file parsable by fparser
+    x90_analyser = X90Analyser()
+    x90_analyser._config = BuildConfig('proj', fab_workspace=tmp_path)
+    x90_analyser._config._prep_output_folders()
+    x90_analyser.run(parsable_x90_path)
+
+    # ensure the files are as expected
+    assert filecmp.cmp(parsable_x90_path, EXPECT_PARSABLE_X90)
+
+    # don't leave this in my git repo
+    unlink(parsable_x90_path)
```

Comment:
Is this left over debug code or is it useful? If the latter a more indicative comment may help.